### PR TITLE
Fast

### PR DESCRIPTION
**Pull request by yijunyu**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/srcml.rb [hunk 1]
```diff
@@ -0,0 +1,144 @@
+# Documentation: http://docs.brew.sh/Formula-Cookbook.html
+#                http://www.rubydoc.info/github/Homebrew/brew/master/Formula
+# PLEASE REMOVE ALL GENERATED COMMENTS BEFORE SUBMITTING YOUR PULL REQUEST!
```
Comment (bfontaine) [inline Formula/srcml.rb]: As stated: please remove these comments.

File: Formula/fast.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  depends_on "protobuf" => :build
+  depends_on "srcml" => :optional
+
+  def install
+    system "cmake", "-G", "Unix Makefiles", *std_cmake_args
+    system "make", "install"
+  end
+
+  test do
+    system "true"
```
Comment (bfontaine) [inline Formula/fast.rb]: We need a test here.

File: Formula/fast.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+class Fast < Formula
+  desc "Flattening Abstract Syntax Trees"
+  homepage "https://github.com/yijunyu/fast"
+  url "https://github.com/yijunyu/fast/archive/v0.0.1.tar.gz"
+  version "0.0.2"
```
Comment (bfontaine) [inline Formula/fast.rb]: Why is this 0.0.2 here but 0.0.1 in the URL?

File: Formula/srcml.rb [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+class Srcml < Formula
+  desc "srcML - an infrastructure for the exploration, analysis, and manipulation of source code"
```
Comment (bfontaine) [inline Formula/srcml.rb]: Please remove the formula name from its description.

File: Formula/srcml.rb [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+  homepage "http://www.srcml.org/"
+  url "http://131.123.42.38/lmcrs/beta/srcML-src.tar.gz"
```
Comment (bfontaine) [inline Formula/srcml.rb]: We need a versioned URL here.

File: Formula/srcml.rb [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+  depends_on "cmake" => :build
+  depends_on "LibArchive"
+  depends_on "libxml2"
+
+#  env do
+#    unless HOMEBREW_PREFIX.to_s == "/usr/local"
+#      ENV.append_path "HOMEBREW_LIBRARY_PATHS", "/usr/local/lib"
+#    end
+#  end
```
Comment (bfontaine) [inline Formula/srcml.rb]: Please remove these comments.

File: Formula/srcml.rb [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+
+  def install
+    #system "cmake", "-DINSTALL_PREFIX=/usr/local/Cellar/srcml/0.9.5", "-G", "Unix Makefiles", *std_cmake_args
```
Comment (bfontaine) [inline Formula/srcml.rb]: Please remove this comment.

File: Formula/srcml.rb [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+    system "cmake", *std_cmake_args
+    system "make", "install", "prefix=#{prefix}"
+  end
+
+  patch :DATA 
+
+  test do
+    system "true"
```
Comment (bfontaine) [inline Formula/srcml.rb]: We need a test here too.

File: Formula/srcml.rb [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+#    end
+#  end
+
+  def install
+    #system "cmake", "-DINSTALL_PREFIX=/usr/local/Cellar/srcml/0.9.5", "-G", "Unix Makefiles", *std_cmake_args
+    system "cmake", *std_cmake_args
+    system "make", "install", "prefix=#{prefix}"
+  end
+
+  patch :DATA
```
Comment (bfontaine) [inline Formula/srcml.rb]: Please submit this patch upstream and add a link to the upstream issue report in a comment.

**Review by bfontaine — CHANGES_REQUESTED** [review]

**Comment by bfontaine** [comment]
This appears to be a self-submission of a relatively new and not well-known project. You can [maintain a formula in your own tap](https://github.com/Homebrew/brew/blob/master/docs/How-to-Create-and-Maintain-a-Tap.md) and re-submit it again here once it’s popular/mature enough.
It’s also easier to review if you make one pull-request per new formula and label each one "`<formula name> <version> (new formula)`", e.g. "`fast 0.0.1 (new formula)`". Thanks!

**Comment by yijunyu** [comment]
Thank you for the comments, I will try a new tap then, thank you for taking a look at it, which has been a lot of learning experience for me. When we get more users I will then ask your help again.

**Pull request closed without merging**